File: tests_wraps/src/main/java/com/github/klee0kai/test/car/di/qualifiers/qualifiers/MyQualifierMulti.java
package com.github.klee0kai.test.car.di.qualifiers.qualifiers;


import javax.inject.Qualifier;
import java.lang.annotation.Documented;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;


@Qualifier
@Retention(RetentionPolicy.RUNTIME)
@Documented
public @interface MyQualifierMulti {

    String id() default "";

    int indx() default 0;

    Type type() default Type.SIMPLE;

    enum Type {
        SIMPLE,
        MIDDLE,
        HARD,
    }
}

File: tests_wraps_kotlin/src/main/kotlin/com/github/klee0kai/test/car/di/qualifiers/qualifiers/MyQualifierMulti.kt
package com.github.klee0kai.test.car.di.qualifiers.qualifiers

import javax.inject.Qualifier

@Qualifier
@Retention(AnnotationRetention.RUNTIME)
@MustBeDocumented
annotation class MyQualifierMulti(val id: String = "", val indx: Int = 0, val type: Type = Type.SIMPLE) {
    enum class Type {
        SIMPLE,
        MIDDLE,
        HARD
    }
}
